rust-mode.el

```diff
@@ -1746,6 +1748,27 @@ visit the new file."
       (let ((output (json-read)))
         (cdr (assoc-string "root" output))))))
 
+(defun rust-insert-dbg ()
+  "Insert the dbg! macro."
+  (insert-parentheses (unless (region-active-p) 1))
+  (backward-char 1)
+  (insert "dbg!"))
+
+;;;###autoload
+(defun rust-dbg-wrap-or-unwrap ()
+  "Either remove or add the dbg! macro."
+  (interactive)
+  (save-excursion
+    (if (region-active-p)
+        (rust-insert-dbg)
+
+      (goto-char (beginning-of-thing 'symbol))
```

Comment:
There's `rust-beginning-of-defun`. I'll probably need to implement `rust-beginning-of-symbol`

Reply:
I feel like I'm reinventing the wheel when trying to implement this function. I'll probably have some bugs that `beginning-of-thing` knows how to handle. Since it's a built-in package what are the reasons not to use it?

Reply:
a problem with your new implementation: it may look backward too far. 

for example:
`dbg!(foo)   foo|`   =>  `foo   foo| `  instead of                `  dbg!(foo)   dbg!(foo|)`


Also note that `dbg!|(foo)` is currently an error (but it may be the fault of the syntax table of rust-mode).

Reply:
I would say that if no already-defined function does the job, don't bother. I thought that you could use something like `(rust-looking-back-str "dbg!(")` but maybe it is not possible.